google-cloud-storage/src/test/java/com/google/cloud/storage/it/ITStorageTest.java

```diff
@@ -1317,6 +1318,55 @@ public void testGetBlobs() {
     assertTrue(remoteBlobs.get(1).delete());
   }
 
+  @Test
+  public void testDownloadPublicBlobWithoutAuthentication() {
+    // create an unauthorized user
+    Storage unauthorizedStorage = StorageOptions.getUnauthenticated().getService();
+
+    // try to download blobs from a public bucket
+    String landsatBucket = "gcp-public-data-landsat";
+    String landsatPrefix = "LC08/PRE/044/034/LC80440342016259LGN00/";
+    String landsatBlob = landsatPrefix + "LC80440342016259LGN00_MTL.txt";
+    byte[] bytes = unauthorizedStorage.readAllBytes(landsatBucket, landsatBlob);
+    assertEquals(7903, bytes.length);
+    int numBlobs = 0;
+    Iterator<Blob> blobIterator = unauthorizedStorage
+      .list(landsatBucket, Storage.BlobListOption.prefix(landsatPrefix))
+      .iterateAll().iterator();
+    while (blobIterator.hasNext()) {
+      numBlobs++;
+      blobIterator.next();
+    }
+    assertEquals(13, numBlobs);
```

Comment:
@garrettjonesgoogle @andreamlin Is there guidance for whether we use Truth or JUnit for asserts?

For new tests, I think Truth makes sense, since it gives better error messages.

Reply:
Let's go with Truth for new tests.